src/status.rs

```diff
@@ -309,9 +427,370 @@ mod test {
+            false,
+            "Date older than the recency limit is not recent"
+        )
+    }
+
+    // is_initializing()
+    #[test]
+    fn two_empty_message_arrays() {
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Both leader heartbeats + newstate and follower heatbeats + approvestate pairs are empty arrays"
+        )
+    }
+
+    #[test]
+    fn leader_has_no_messages() {
+        let approve_state = get_approve_state_msg();
+        let heartbeat = get_heartbeat_msgs(Duration::minutes(0));
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Leader has no new messages but the follower has heartbeats and approvestate"
+        )
+    }
+
+    #[test]
+    fn follower_has_no_messages() {
+        let heartbeat = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Follower has no new messages but leader has heartbeats and newstate"
+        )
+    }
+
+    #[test]
+    fn both_arrays_have_messages() {
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(0));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
```

Comment:
Is this the same test from the Market?
https://github.com/AdExNetwork/adex-market/blob/master/test/validatorTestMessages.js#L165

If so, they are a bit different, maybe you can elaborate on that?

Reply:
Yes, the initializing state is when `there are no messages at all for at least one validator`. This is the case where both validators have messages therefore it's false. The difference is that here there is also a newState and an approveState. I will add another with only heartbeats (both should be false)